src/libraries/Common/src/System/Reflection/AssemblyNameParser.cs

```diff
@@ -0,0 +1,568 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+#nullable enable
+
+namespace System.Reflection
+{
+    //
+    // Parses an assembly name.
+    //
+    internal ref partial struct AssemblyNameParser
+    {
+        public readonly struct AssemblyNameParts
+        {
+            public AssemblyNameParts(string name, Version? version, string? cultureName, AssemblyNameFlags flags, byte[]? publicKeyOrToken)
+            {
+                _name = name;
+                _version = version;
+                _cultureName = cultureName;
+                _flags = flags;
+                _publicKeyOrToken = publicKeyOrToken;
+            }
+
+            public readonly string _name;
+            public readonly Version? _version;
+            public readonly string? _cultureName;
+            public readonly AssemblyNameFlags _flags;
+            public readonly byte[]? _publicKeyOrToken;
+        }
+
+        // Token categories for the lexer.
+        private enum Token
+        {
+            Equals = 1,
+            Comma = 2,
+            String = 3,
+            End = 4,
+        }
+
+        private enum AttributeKind
+        {
+            Version = 1,
+            Culture = 2,
+            PublicKeyOrToken = 4,
+            ProcessorArchitecture = 8,
+            Retargetable = 16,
+            ContentType = 32
+        }
+
+        private readonly ReadOnlySpan<char> _input;
+        private readonly bool _strict;
```

Comment:
The strict mode is not part of the approved surface. Delete this?

Reply:
I've allowed myself to keep it, as I expect that this feature will be back very soon, but just expressed with a different public API. It's simply less work for me to keep it, rather than remove it now and re-add it very soon.

But.. this is my laziness speaking. Please let me know if I should remove it.

Reply:
It is impossible to review it when it is not specified what it is supposed to do.

(Also, `strict` is not a good name for it as was discussed during API review.)

Reply:
> It is impossible to review it when it is not specified what it is supposed to do.

This is a perspective I was lacking. I am going to remove it, apologies for trouble!